### ISSUE TITLE
combining length_bytes! with map!

### ISSUE THREAD

**Issue by GGist** [Geal/nom#236]
I ran in to an issue with ```length_bytes!``` failing to parse correctly when ```map!``` is embedded directly as the parser combinator.

Failing Parser:
```
length_bytes!(map!(be_u8, |b| b as usize))
```

Build Error:
```
src\option.rs:101:56: 101:57 error: expected ident, found |
src\option.rs:101             option_contents: length_bytes!(map!(be_u8, |b| b as usize)) ~
                                                                         ^
Could not compile `bip_utracker`.

To learn more, run the command again with --verbose.
```

Working Parser:
```
length_bytes!(byte_usize)

named!(byte_usize<&[u8], usize>, map!(be_u8, |b| b as usize));
```

I am running nom 1.2.2 with the following:
rustc 1.8.0-nightly (c92e910c1 2016-02-21)
binary: rustc
commit-hash: c92e910c117775563a6a8782db16a15215061bc1
commit-date: 2016-02-21
host: x86_64-pc-windows-msvc
release: 1.8.0-nightly

**Comment by Geal** [comment]
Hi!
with nom's master, it builds correctly, so this bug will be fixed once I release nom 1.2.3 (probably this afternoon). Also, there will be no need to convert the result of `be_u8` to `usize`,  `length_bytes` does it internally now.

**Comment by GGist** [comment]
Awesome, much appreciated!
